File: Frontend/types/berita.ts
export type KategoriBerita = 'pengumuman' | 'kegiatan' | 'pembangunan' | 'lainnya';

export interface Berita {
  id: string;
  slug: string;
  judul: string;
  ringkasan: string;
  konten: string;
  kategori: KategoriBerita;
  gambarSampul: string;
  gambarTengah?: string;
  penulis: string;
  tanggalTerbit: string;
  tags: string[];
  dibaca: number;
}

export interface BeritaKomentar {
  id: string;
  beritaId: string;
  userId?: string | null;
  nama: string;
  konten: string;
  createdAt: string;
}

export interface BeritaListParams {
  kategori?: KategoriBerita;
  search?: string;
  page?: number;
  perPage?: number;
}

export interface PaginatedResult<T> {
  data: T[];
  total: number;
  page: number;
  perPage: number;
  totalPages: number;
}

File: Backend/internal/domain/berita.go
package domain

import "time"

type BeritaStatus string

const (
	BeritaDraft  BeritaStatus = "draft"
	BeritaTerbit BeritaStatus = "terbit"
)

type Berita struct {
	ID, Slug, Judul, Ringkasan, Konten string
	Kategori                           string
	GambarSampul                       string
	GambarTengah                       string // opsional — tampil di tengah artikel
	PenulisID                          *string
	Penulis                            string
	Tags                               []string
	Status                             BeritaStatus
	TanggalTerbit                      *time.Time
	Dibaca                             int
	CreatedAt, UpdatedAt               time.Time
}

// BeritaKomentar adalah komentar publik pada sebuah artikel berita.
type BeritaKomentar struct {
	ID        string
	BeritaID  string
	UserID    *string
	Nama      string
	Konten    string
	CreatedAt time.Time
}
